egui/src/widgets/plot/items.rs

```diff
@@ -1446,3 +1560,947 @@ impl PlotItem for PlotImage {
+                }
+            } else {
+                let mut min = 0.0;
+                for other_chart in others {
+                    if let Some(other_bar) = other_chart.bars.get(index) {
+                        let other_lower = other_bar.lower();
+                        if other_lower < min {
+                            min = other_lower;
+                        }
+                    }
+                }
+                if min.abs() > 0.0 {
+                    bar.base_offset = Some(min);
+                }
+            }
+        }
+        self
+    }
+}
+
+impl PlotItem for BarChart {
+    fn get_shapes(&self, _ui: &mut Ui, transform: &ScreenTransform, shapes: &mut Vec<Shape>) {
+        self.bars.iter().for_each(|b| {
+            b.add_shapes(transform, self.highlight, shapes);
+        });
+    }
+
+    fn initialize(&mut self, _x_range: RangeInclusive<f64>) {
+        // nothing to do
+    }
+
+    fn name(&self) -> &str {
+        self.name.as_str()
+    }
+
+    fn color(&self) -> Color32 {
+        self.default_color
+    }
+
+    fn highlight(&mut self) {
+        self.highlight = true;
+    }
+
+    fn highlighted(&self) -> bool {
+        self.highlight
+    }
+
+    fn geometry(&self) -> PlotGeometry<'_> {
+        PlotGeometry::Rects
+    }
+
+    fn get_bounds(&self) -> Bounds {
+        let mut bounds = Bounds::NOTHING;
+        self.bars.iter().for_each(|b| {
+            bounds.merge(&b.bounds());
+        });
+        bounds
+    }
+
+    fn find_closest(&self, point: Pos2, transform: &ScreenTransform) -> Option<HoverElement> {
+        find_closest_rect(&self.bars, point, transform)
+    }
+
+    fn on_hover(&self, elem: HoverElement, shapes: &mut Vec<Shape>, plot: &PlotConfig<'_>) {
+        let bar = &self.bars[elem.index];
+
+        bar.add_shapes(plot.transform, true, shapes);
+        bar.add_rulers_and_text(self, plot, shapes);
+    }
+}
+
+/// A box in a [`BoxPlot`] diagram. This is a low level graphical element; it will not compute quartiles and whiskers,
+/// letting one use their preferred formula. Use [`Points`] to draw the outliers.
+#[derive(Clone, Debug, PartialEq)]
+pub struct BoxElem {
+    /// Name of plot element in the diagram (annotated by default formatter).
+    pub name: String,
+
+    /// Which direction the box faces in the diagram.
+    pub orientation: Orientation,
+
+    /// Position on the argument (input) axis -- X if vertical, Y if horizontal.
+    pub argument: f64,
+
+    /// Value of lower whisker (typically minimum).
+    ///
+    /// The whisker is not drawn if `lower_whisker >= quartile1`.
+    pub lower_whisker: f64,
+
+    /// Value of lower box threshold (typically 25% quartile)
+    pub quartile1: f64,
+
+    /// Value of middle line in box (typically median)
+    pub median: f64,
+
+    /// Value of upper box threshold (typically 75% quartile)
+    pub quartile3: f64,
+
+    /// Value of upper whisker (typically maximum)
+    pub upper_whisker: f64,
```

Comment:
Perhaps these should be combined into a `[f64; 5]` here and in `new`. This communicates how they are all closely related (on the same dimension, and should be ordered in an increasing order). Or they can be extracted to a `struct BoxSpread` to keep down the complexity of `BoxElem`.